homeassistant/components/dlna_dms/dms.py

```diff
@@ -71,26 +65,19 @@ def __init__(
         session = aiohttp_client.async_get_clientsession(hass, verify_ssl=False)
         self.requester = AiohttpSessionRequester(session, with_sleep=True)
         self.upnp_factory = UpnpFactory(self.requester, non_strict=True)
-        # NOTE: event_handler is not actually used, and is only created to
-        # satisfy the DmsDevice.__init__ signature
-        self.event_handler = UpnpEventHandler(UpnpNotifyServer(), self.requester)
         self.devices = {}
         self.sources = {}
 
     async def async_setup_entry(self, config_entry: ConfigEntry) -> bool:
         """Create a DMS device connection from a config entry."""
         assert config_entry.unique_id
         async with self.lock:
```

Comment:
Why do we need a lock here?

Reply:
Good question. It was a hold-over from several revisions ago and can be removed.